lib/java/js-websocket/src/main/resources/org/enso/polyfill/websocket/websocket-polyfill.js

```diff
@@ -0,0 +1,282 @@
+(function (jvm) {
+
+    //
+    // EventBase
+    //
+
+    var EventBase = {
+        bubbles: false,
+        cancelable: false
+    };
+
+    //
+    // WebSocket
+    //
+
+    var WebSocket = (function () {
```

Comment:
Unless ES6+ code isn't supported directly, you could be using a `class` here, instead of using IIFE, calling `defineProperty` or manually assinging methods to the prototype. As it stands, this code looks more like an output of a bundler targetting old JS versions.

Reply:
> this code looks more like an output of a bundler targetting old JS versions.

That's the way [I write JavaScript](https://github.com/JaroslavTulach/graaljs-websocket/blob/3c592f0efd48a6320998951564447f4b91ca4444/src/main/java/org/apidesign/polyfill/websocket/WebSocketPolyfill.java#L26) to make it portable and to know what exactly it does.

> Unless ES6+ code isn't supported directly, you could be using a `class` here, 

[Graal.js](https://github.com/oracle/graaljs) implements [ECMAScript 2023 specification](https://262.ecma-international.org/14.0/). Using `class` is possible, but: with this old fashion code style we do _control what is visible_ to clients of the API.  We certainly _don't want the `jvm` variable to be visible_ to JavaScript code outside of the enclosing `function`.

Suggestions to rewrite this code to look more modern and still keep the encapsulation are welcomed.

Reply:
The classes give you encapsulation just fine, and also allow you to use [privates](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties) if you so wish. You also still control what's visible to the clients on the module level. Ideally that would be done through ES module exports, but you can still keep using IIFE and assigning to global as usual if there is any problem with using modules. Just moving away from manual prototype assinments to class would be a good step.